packages/table/src/Row/InternalRowWithRT.tsx

```diff
@@ -80,7 +80,13 @@ export const MemoizedInternalRowWithRT = React.memo(
     const { children: prevChildren, ...restPrevProps } = prevProps;
     const { children: nextChildren, ...restNextProps } = nextProps;
 
-    const propsAreEqual = isEqual(restPrevProps, restNextProps);
+    // This allows us to rerender if the child(cell) count changes. E.g. column visibility changes
+    const prevChildrenCount = React.Children.count(prevChildren);
+    const nextChildrenCount = React.Children.count(nextChildren);
+
+    const propsAreEqual =
+      isEqual(restPrevProps, restNextProps) &&
+      prevChildrenCount === nextChildrenCount;
```

Comment:
Would checking `isEqual` on _all_ props work?

Reply:
No, it doesn't.
```js
console.log({
  prevChildren,
  nextChildren,
  childrenAreEqual: isEqual(prevChildren, nextChildren), // false
  propsAreEqual: isEqual(restPrevProps, restNextProps), //true
  everythingIsEqual: isEqual(prevProps, nextProps), //false
});

const propsAreEqual = isEqual(prevProps, nextProps);
```
 [isEqual](https://commerce.nearform.com/blog/2018/introducing-react-fast-compare) should be doing a deep comparison check, so something must differ between the children.